src/modules/fancyzones/editor/FancyZonesEditor/Models/LayoutModel.cs

```diff
@@ -250,6 +288,12 @@ public static ObservableCollection<LayoutModel> LoadCustomModels()
                             i++;
                         }
 
+                        if (error)

```

Comment:
Perhaps we should also check if `cellChildMap` is valid:

There's some number k >= 1 such that cellChildMap contains exactly the set {0, 1, ..., k-1}
For each x in this set, the elements of cellChildMap equal to x should form a rectangular submatrix.

Because this algorithm is not exactly trivial, it should probably be written as a separate function.

Reply:
That's true. Intention of this PR was to prevent editor crashes, which won't happen in that case (it would just be some weird grid layout). So that should probably be checked by fancyzoneslib